smartsim/entity/param_data_class.py

```diff
@@ -0,0 +1,11 @@
+from dataclasses import dataclass, field
+
+
+@dataclass
+class ParamSet:
+    """
+    Represents a set of file parameters and execution arguments.
+    """
+
+    params: dict[str, str] = field(default_factory=dict)
+    exe_args: dict[str, list[str]] = field(default_factory=dict)
```

Comment:
If you want to make sure no one does something silly like asigning to one of these attrs
```py
ParamSet(file_params, exe_args).exe_args = {}  # <-- AHH! This should almost certainly
                                               #     not be allowed!!
```
consider marking this dataclass as "frozen"
```suggestion
@dataclass(frozen=True)
class ParamSet:
    """
    Represents a set of file parameters and execution arguments.
    """

    params: dict[str, str] = field(default_factory=dict)
    exe_args: dict[str, list[str]] = field(default_factory=dict)
```

Reply:
ohhh I see